modules/k8s/synapse-ha/main.tf

```diff
@@ -0,0 +1,164 @@
+terraform {
+  required_version = ">= 1.6.6"
+  required_providers {
+    juju = {
+      source  = "juju/juju"
+      version = ">= 0.10.1"
+    }
+  }
+}
+
+resource "juju_model" "synapse" {
+  name = var.juju_model_name
+  config = {
+    juju-http-proxy  = "${var.proxy_scheme}://%{if var.proxy_user != "" && var.proxy_pass != ""}${var.proxy_user}:${var.proxy_pass}@%{endif}${var.proxy_hostname}:${var.proxy_port}"
+    juju-https-proxy = "${var.proxy_scheme}://%{if var.proxy_user != "" && var.proxy_pass != ""}${var.proxy_user}:${var.proxy_pass}@%{endif}${var.proxy_hostname}:${var.proxy_port}"
+    juju-no-proxy    = "127.0.0.1,localhost,::1,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,.canonical.com,.launchpad.net,.internal,.jujucharms.com,.ubuntu.com"
+  }
+  constraints = "root-disk-source=volume"
+  cloud {
+    name   = var.cloud_name
+    region = "default"
+  }
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+
+resource "juju_application" "synapse" {
```

Comment:
Even considering S3, I would add a constraint here to deploy Synapse already with a storage bigger than the default (1GB).  :smiling_face_with_tear:

Reply:
I thought that Juju terraform provider already had storage support to application resource but this feature is only available in the release candidate:
https://github.com/juju/terraform-provider-juju/releases/tag/0.13.0-rc1

So, NVM for now :( Maybe we can get back to this once is released.